src/www/app/util.ts

```diff
@@ -14,14 +14,14 @@
 
 import * as errors from '../model/errors';
 
-export function timeoutPromise<T>(promise: Promise<T>, ms: number, name = ''): Promise<T> {
-  let winner: Promise<T>;
+export function timeoutPromise<T>(promise: Promise<T>, ms: number, name = ''): Promise<T | undefined> {
```

Comment:
I found this a little odd and almost rewrote it but... I _think_ this is more correct? it was complaining about `winner` needing to be a const before because it wasn't assigned more than once.

Reply:
You might prefer `Promise<T | void>` ... but really I think this should be rewritten.